xopt/generators/bayesian/bax_generator.py

```diff
@@ -0,0 +1,123 @@
+# +
+import logging
+import time
+from typing import Callable, Dict, List
+
+from botorch.optim import optimize_acqf
+from botorch.optim.initializers import sample_truncated_normal_perturbations
+
+from xopt.generators.bayesian.bax.acquisition import ExpectedInformationGain
+
+from xopt.generators.bayesian.bayesian_generator import BayesianGenerator
+from xopt.generators.bayesian.objectives import create_mc_objective
+from xopt.generators.bayesian.options import BayesianOptions
+from xopt.vocs import VOCS
+
+logger = logging.getLogger()
+
+
+class BAXGenerator(BayesianGenerator):
+    alias = "BAX"
+
+    def __init__(
+        self,
+        vocs: VOCS,
+        meas_param: str,
+        algo_class: Callable,
+        algo_kwargs: Dict,
+        options: BayesianOptions = None,
```

Comment:
Its time to put all of these options in the correct pydantic structure, I'll reach out so we can discuss